examples/stdlib/io/promise.effekt

```diff
@@ -1,12 +1,13 @@
 import io
 
-def main() = {
-  val p = promise::make[Int]()
+def main() = eventloop(box {
 
-  promise::await(p, box { n => println("Resolved 1: " ++ show(n)) })
-  promise::await(p, box { n => println("Resolved 2: " ++ show(n)) })
-  promise::await(p, box { n => println("Resolved 3: " ++ show(n)) })
+  val p = promise::make[String]()
 
-  promise::resolve(p, 42)
+  spawn(box { println("Resolved 1: " ++ promise::await(p)) });
+  spawn(box { println("Resolved 2: " ++ promise::await(p)) });
+  spawn(box { println("Resolved 3: " ++ promise::await(p)) });
 
-}
\ No newline at end of file
+  promise::resolve(p, "42")
+
+})
```

Comment:
In the Effekt meeting, we said that we want to always automatically insert libuv::start if `main` includes capture `control`.